metals/src/main/scala/scala/meta/internal/metals/testProvider/frameworks/MunitTestFinder.scala

```diff
@@ -0,0 +1,188 @@
+package scala.meta.internal.metals.testProvider.frameworks
+
+import scala.annotation.tailrec
+import scala.collection.immutable
+import scala.collection.mutable
+
+import scala.meta.Defn
+import scala.meta.Lit
+import scala.meta.Pkg
+import scala.meta.Term
+import scala.meta.Tree
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.testProvider.FullyQualifiedName
+import scala.meta.internal.metals.testProvider.TestCaseEntry
+import scala.meta.internal.parsing.Trees
+import scala.meta.internal.semanticdb.SymbolOccurrence
+import scala.meta.internal.semanticdb.TextDocument
+import scala.meta.io.AbsolutePath
+
+class MunitTestFinder(trees: Trees) {
+
+  def findTests(
+      doc: TextDocument,
+      path: AbsolutePath,
+      suiteName: FullyQualifiedName
+  ): Vector[TestCaseEntry] = {
+    val uri = path.toURI
+    val testcases = new mutable.ArrayBuffer[TestCaseEntry]()
+    val occurences = doc.occurrences
+      .filter(occ =>
+        occ.symbol.startsWith("munit/") && occ.symbol.contains("test")
```

Comment:
I don;t have a better idea how to check if symbol comes from munit and it is a `test` method, this seems to work

Reply:
What are the symbols that we match here?